core/cas-server-core-configuration-api/src/main/java/org/apereo/cas/configuration/DefaultCasConfigurationPropertiesSourceLocator.java

```diff
@@ -39,37 +33,67 @@
-        val propertyNames = String.join("|", profiles);
-        val profiledProperties = profiles.stream()
-            .map(p -> String.format("application-%s", p))
-            .collect(Collectors.joining("|"));
+    // will be processed in order, last one wins
+    private List<File> getAllPossibleExternalConfigDirFilenames(final File configdir, final List<String> profiles) {
+        val fileNames = new ArrayList<File>();
+        for (val appName: getApplicationNames()) {
+            for (val ext: EXTENSIONS) {
+                fileNames.add(new File(configdir, String.format("%s.%s", appName, ext)));
+            }
+        }
+        for (val profile: profiles) {
+            for (val ext: EXTENSIONS) {
+                fileNames.add(new File(configdir, String.format("application-%s.%s", profile, ext)));
+                // spring boot normally doesn't do <profile>.<ext>, only application-<profiles>.<ext>
+                fileNames.add(new File(configdir, String.format("%s.%s", profile, ext)));
+            }
+        }
+        return fileNames;
+    }
 
-        val regex = String.format("(%s|%s|application)\\.(yml|properties)", propertyNames, profiledProperties);
-        LOGGER.debug("Looking for configuration files at [{}] that match the pattern [{}]", config, regex);
-        return regex;
+    // return files that exist
+    private List<File> scanForConfigurationFiles(final File config, final List<String> profiles) {
+        val possibleFiles = getAllPossibleExternalConfigDirFilenames(config, profiles);
+        return possibleFiles.stream().filter(File::exists).filter(File::isFile).collect(Collectors.toList());
     }
 
+    /**
+     * Adding items to composite property source which contains property sources processed in order
+     * where first one containing the property is used:
+     * First Priority: Standalone configuration file
+     * Second Priority: Configuration files in config dir, profiles override non-profiles, last profile overrides first
+     * Third Priority: classpath:/application.yml, although spring boot may have already processed this?
+     * @param environment    the environment
+     * @param resourceLoader the resource loader
+     * @return CompositePropertySource containing sources listed above
+     */
     @Override
     public PropertySource<?> locate(final Environment environment, final ResourceLoader resourceLoader) {
         val compositePropertySource = new CompositePropertySource("casCompositePropertySource");
 
+        // Get properties from specified single config file.
         val configFile = casConfigurationPropertiesEnvironmentManager.getStandaloneProfileConfigurationFile();
         if (configFile != null) {
             val sourceStandalone = loadSettingsFromStandaloneConfigFile(configFile);
             compositePropertySource.addPropertySource(sourceStandalone);
         }
 
+        // Get files from config directory
```

Comment:
While I appreciate you taking the time to explain things, this sort of thing is rather obvious and doesn't exactly need a comment perhaps?